src/node_jsvmapi.cc

```diff
@@ -514,6 +512,9 @@ namespace v8impl {
 #define CHECK_NEW_FROM_UTF8(isolate, result, str)                       \
     CHECK_NEW_FROM_UTF8_LEN((isolate), (result), (str), -1)
 
+#define GET_RETURN_STATUS()                                             \
+  (tryCatch.HasCaught() ? napi_ok : napi_pending_exception)
+
```

Comment:
Good catch on the condition. I'll add a call to napi_set_last_error only in the pending exception case. The stored error code should still be napi_ok if there was no previous error.